docs/reference/analysis/tokenfilters/synonym-graph-tokenfilter.asciidoc

```diff
@@ -85,45 +85,42 @@ Additional settings are:
 <<indices-reload-analyzers,reloading>> search analyzers to pick up
 changes to synonym files. Only to be used for search analyzers.
 * `expand` (defaults to `true`).
-* `lenient` (defaults to `false`). If `true` ignores exceptions while parsing the synonym configuration. It is important
-to note that only those synonym rules which cannot get parsed are ignored. For instance consider the following request:
-
-[source,console]
---------------------------------------------------
-PUT /test_index
-{
-  "settings": {
-    "index": {
-      "analysis": {
-        "analyzer": {
-          "synonym": {
-            "tokenizer": "standard",
-            "filter": [ "my_stop", "synonym_graph" ]
-          }
-        },
-        "filter": {
-          "my_stop": {
-            "type": "stop",
-            "stopwords": [ "bar" ]
-          },
-          "synonym_graph": {
-            "type": "synonym_graph",
-            "lenient": true,
-            "synonyms": [ "foo, bar => baz" ]
-          }
-        }
-      }
-    }
-  }
-}
---------------------------------------------------
+Expand definitions for equivalent synonyms mappings.
+See <<synonym-graph-tokenizer-expand-equivalent-synonyms,expand equivalent synonyms>>.
+* `lenient` (defaults to `false`).
+If `true` ignores errors while parsing the synonym configuration.
+It is important to note that only those synonym rules which cannot get parsed are ignored.
+See <<synonym-graph-tokenizer-stop-token-filter,synoyms and stop token filters>> for an example of `lenient` behaviour for invalid synonym rules.
+
+[discrete]
+[[synonym-graph-tokenizer-expand-equivalent-synonyms]]
+===== `expand` equivalent synonyms
+
+The `expand` parameter controls whether to expand equivalent synonyms mappings.
+Consider a synonym defined like:
+
+`foo, bar, baz`
 
-With the above request the word `bar` gets skipped but a mapping `foo => baz` is still added. However, if the mapping
-being added was `foo, baz => bar` nothing would get added to the synonym list. This is because the target word for the
-mapping is itself eliminated because it was a stop word. Similarly, if the mapping was "bar, foo, baz" and `expand` was
-set to `false` no mapping would get added as when `expand=false` the target mapping is the first word. However, if
-`expand=true` then the mappings added would be equivalent to `foo, baz => foo, baz` i.e, all mappings other than the
-stop word.
+Using `expand: true` would get the synonym expanded into:
+
+```
+foo => bar
+foo => baz
+bar => foo
+bar => baz
+baz => foo
+baz => bar
+```
```

Comment:
This representation misses that the "foo" representation of the synonym would also remain. I also think that this representation makes it unclear what the actual token output would be when analysis is performed. Maybe we use the analyze API output format (or something similar to that) to show how a token "foo" would be analyzed in this case?

Reply:
Just tried with this test analyzer:

```json
PUT test-index
{
    "settings": {
        "number_of_shards": 1,
        "number_of_replicas": 0,
        "analysis": {
            "analyzer": {
                "synonym_analyzer": {
                    "tokenizer": "whitespace",
                    "filter": [
                        "synonym_filter"
                    ]
                }
            },
            "filter": {
                "synonym_filter": {
                  "type": "synonym_graph",
                  "synonyms": [
                    "foo, bar, baz"
                  ]
                }
            }
        }
    }
}
```

Running through the `_analyze` API:

```json
GET test-index/_analyze
{
  "analyzer": "synonym_analyzer",
  "text": "foo"
}

{
    "tokens": [
        {
            "token": "bar",
            "start_offset": 0,
            "end_offset": 3,
            "type": "SYNONYM",
            "position": 0
        },
        {
            "token": "baz",
            "start_offset": 0,
            "end_offset": 3,
            "type": "SYNONYM",
            "position": 0
        },
        {
            "token": "foo",
            "start_offset": 0,
            "end_offset": 3,
            "type": "word",
            "position": 0
        }
    ]
}
```

My point is that the `foo` token is retained, which is not accurately represented by the explicit synonyms in this code block. It's confusing because you're reusing the `=>` delimiter, which is used earlier in this doc to define an explicit synonym. As the reader, I assume you are defining a set of explicit synonyms. Was that your intent?

Reply:
To be clear, I think it's fine to use the explicit synonym format (with `=>`), but we should be sure that the explicit synonyms accurately represent the synonym expansion behavior

Reply:
I see - you're right, let's add the explicit mappings for the identity expansions 👍